httpserver/oauth2_provider.go

```diff
@@ -0,0 +1,101 @@
+package httpserver
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"time"
+
+	"github.com/ory/fosite"
+	"github.com/ory/fosite/compose"
+	"github.com/ory/fosite/handler/openid"
+	"github.com/ory/fosite/token/jwt"
+)
+
+// OAuth2Provider manages OAuth2 operations
+type OAuth2Provider struct {
+	oauth2  fosite.OAuth2Provider
+	storage *OAuth2Storage
+	config  *fosite.Config
+}
+
+// NewOAuth2Provider creates a new OAuth2 provider with SQLite storage
+func NewOAuth2Provider(dbPath string, issuer string) (*OAuth2Provider, error) {
+	storage, err := NewOAuth2Storage(dbPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create storage: %w", err)
+	}
+
+	// Generate RSA key for JWT signing
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
```

Comment:
Persist the private key in the database.  Prefer to load it from the database and only generate a new one if it isn't present.